File: android/src/main/java/org/battleshipgame/android/activities/MainActivity.java
package org.battleshipgame.android.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import org.battleshipgame.android.GameInstance;
import org.battleshipgame.android.R;

public class MainActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        GameInstance.create();
    }

    public void startNewGame(View view) {
        Intent intent = new Intent(this, GameModeActivity.class);
        startActivity(intent);
    }

    public void exit(View view) {
        finish();
    }
}

File: android/src/main/java/org/battleshipgame/android/activities/MainActivity.kt
package org.battleshipgame.android.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import org.battleshipgame.android.GameInstance
import org.battleshipgame.android.R

class MainActivity : StandardActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        GameInstance.create()
    }

    fun startNewGame(view: View) {
        val intent = Intent(this, GameModeActivity::class.java)
        startActivity(intent)
    }

    fun exit(view: View?) {
        finish()
    }
}